cnf-certification-test/networking/suite.go

```diff
@@ -259,3 +266,58 @@ func testDualStackServices(env *provider.TestEnvironment) {
 		ginkgo.Fail(fmt.Sprintf("Found %d services in error, check error log for details", len(errorServices)))
 	}
 }
+
+//nolint:funlen
+func testNetworkPolicyDenyAll(env *provider.TestEnvironment) {
+	ginkgo.By("Test for Deny All in network policies")
+	var namespacesMissingDenyAllDefaultPolicies []string
+	var processedNamespaces []string
+
+	for _, put := range env.Pods {
+		denyAllEgressFound := false
+		denyAllIngressFound := false
+
+		// Short circuit if the namespace has already been processed
+		if stringhelper.StringInSlice(processedNamespaces, put.Data.Namespace, false) {
+			logrus.Debugf("Namespace: %s has already been processed for deny-all network policies.  Skipping.", put.Data.Namespace)
+			break
+		}
+
+		// Look through all of the network policies for a matching namespace.
+		for index := range env.NetworkPolicies {
+			// Match the pod namespace with the network policy namespace.
+			if put.Data.Namespace == env.NetworkPolicies[index].Namespace {
+				// Check to see if the network policy is deny-all (and contains ingress, egress, (or both) network policy types)
+				denyAllExists, foundPolicies := policies.IsNetworkPolicyDenyAll(&env.NetworkPolicies[index])
```

Comment:
I think that a more generic name (and more internal processing) for the function `IsNetworkPolicyDenyAll()` could simplify the code at this top level. For instance, something like `IsNetworkPolicyCompliant()` could return a `bool` and `error`, with the error indicating why it's not compliant. This way all the code to check the different requirements for a Network Policy to be compliant can be hidden inside the function and there is no need to return the policies.

Reply:
Good idea, I'll try to improve.